be/src/util/brpc_stub_cache.h

```diff
@@ -136,8 +127,50 @@ class BrpcStubCache {
     }
 
 private:
+    // StubPool is used to store all stubs with a single endpoint, and the client in the same BE process maintains up to
+    // brpc_max_connections_per_server single connections with each server.
+    // These connections will be created during the first few accesses and will be reused later.
+    struct StubPool {
+        StubPool() { _stubs.reserve(config::brpc_max_connections_per_server); }
+
+        ~StubPool() {
+            for (auto& stub : _stubs) {
+                delete stub;
+            }
+        }
+
+        doris::PBackendService_Stub* get_or_create(const butil::EndPoint& endpoint) {
+            if (UNLIKELY(_stubs.size() < config::brpc_max_connections_per_server)) {
+                brpc::ChannelOptions options;
+                options.connect_timeout_ms = 3000;
+                // Explicitly set the max_retry
+                // TODO(meegoo): The retry strategy can be customized in the future
+                options.max_retry = 3;
+                // the single connection of brpc will only maintain one connection with the same server by default,
+                // all requests are sent on this connection and the throughput will be limited by this.
+                // we use `connection_group` to create multiple single connections to remove this bottleneck.
+                options.connection_group = std::to_string(_stubs.size());
+                std::unique_ptr<brpc::Channel> channel(new brpc::Channel());
+                if (channel->Init(endpoint, &options)) {
+                    return nullptr;
+                }
+                auto stub = new doris::PBackendService_Stub(channel.release(),
```

Comment:
when would you release this object ?

Reply:
> when would you release this object ?

after it is created, it will always exist in the life cycle of BE and will not be released.